docs/reference/sql/data-definition.md

```diff
@@ -0,0 +1,228 @@
+[serialized](../../../overview/apache-kafka-primer/#serializers) independently.
+ksqlDB enables you to exercise this same flexibility and builds on the semantics
+of {{ site.ak }} records, rather than hiding them.
+
+There is no theoretical limit on the number of columns in a stream or table.
+In practice, the limit is determined by the maximum message size that {{ site.ak }}
+can store and the resources dedicated to ksqlDB.
+
+## Streams
+
+A stream is a partitioned, immutable, append-only collection that represents a
+series of historical facts. For example, the rows of a stream could model a
+sequence of financial transactions, like "Alice sent $100 to Bob", followed by
+"Charlie sent $50 to Bob".
+
+Once a row is inserted into a stream, it can never change. New rows can be
+appended at the end of the stream, but existing rows can never be updated or
+deleted.
+
+Each row is stored in a particular partition. Every row, implicitly or explicitly,
+has a key that represents its identity. All rows with the same key reside in the
+same partition.
+
+To create a stream, use the `CREATE STREAM` command. The following example
+statement specifies a name for the new stream, the names of the columns, and
+the data type of each column.
+
+```sql
+CREATE STREAM s1 (
+    k VARCHAR KEY,
+    v1 INT,
+    v2 VARCHAR
+) WITH (
+    kafka_topic = 's1',
+    partitions = 3,
+    value_format = 'json'
+);
+```
+
+This creates a new stream named `s1` with three columns: `k`, `v1`, and `v2`.
+The column `k` is designated as the key of this stream, which controls the
+partition that each row is stored in. When the data is stored, the value
+portion of each row's underlying {{ site.ak }} record is serialized in the
+JSON format.
+
+Under the hood, each stream corresponds to a [Kafka topic](../../../overview/apache-kafka-primer/#topics)
+with a registered schema. If the backing topic for a stream doesn't exist when
+you declare it, ksqlDB creates it on your behalf, as shown in the previous
+example statement.
+
+You can also declare a stream on top of an existing topic. When you do that,
+ksqlDB simply registers its associated schema. If topic `s2` already exists,
+the following statement register a new stream over it:
+
+```sql
+CREATE STREAM s2 (
+    k1 VARCHAR KEY,
+    v1 VARCHAR
+) WITH (
+    kafka_topic = 's2',
+    value_format = 'json'
+);
+```
+
+!!! tip
+    When you create a stream on an existing topic, you don't need to declare
+    the number of partitions for the topic. ksqlDB infers the partition count
+    from the existing topic.
+
+## Tables
+
+A table is a mutable, partitioned collection that models change over time. In
+contrast with a stream, which represents a historical sequence of events, a
+table represents what is true as of "now". For example, you might use a table
+to model the locations where someone has lived as a stream: first Miami, then
+New York, then London, and so forth.
```

Comment:
>  a table represents what is true as of "now"

Hummm... you can also use a table to ask 'what was true as of <some time>'.   So I'm not sure this is a good description of a table.